test/SILOptimizer/sroa_ossa.sil

```diff
@@ -0,0 +1,440 @@
+// RUN: %target-sil-opt -enable-sil-verify-all -sroa %s | %FileCheck %s
```

Comment:
I'd like to add some non-trivial tests. I might do that as its own patch though.